NPE during pointer analysis when using Zipper-e

### 📝 Overall Description

@jjppp reported a NPE, which occurs when Zipper-e is enabled in pointer analysis and when analyzing certain projects (such as jasperreports). 

### 🎯 Expected Behavior

No exceptions.

### 🐛 Current Behavior

NPE:
```
Caused by: java.lang.NullPointerException: Cannot invoke "pascal.taie.util.MutableInt.intValue()" because the return value of "java.util.Map.get(Object)" is null
	at pascal.taie.analysis.pta.toolkit.zipper.Zipper.getPrecisionCriticalMethods(Zipper.java:189)
```
https://github.com/pascal-lab/Tai-e/blob/0f4bcf408de715030080e32f44c60d6aab8f660a/src/main/java/pascal/taie/analysis/pta/toolkit/zipper/Zipper.java#L186-L196

### 🔄 Reproducible Example

jasperreports from https://github.com/jjppp/java-benchmarks/tree/c2953672f9e4096d6e1b2e1c16f86bc7bf8b6696

### ⚙️ Tai-e Arguments

<details>
  <summary>🔍 Click here to see Tai-e Options</summary>
  <!-- Please leave one blank line below for enabling the code block rendering. -->

  ```yaml
  optionsFile: null
printHelp: false
classPath: [
  jasperreports/3.7.4/javaflow.jar,
  jasperreports/3.7.4/lib
]
appClassPath:
- jasperreports/3.7.4/jasperreports.jar
mainClass: net.sf.jasperreports.view.JasperViewer
inputClasses: []
javaVersion: 8
prependJVM: false
allowPhantom: true
worldBuilderClass: pascal.taie.frontend.soot.SootWorldBuilder
outputDir: output
preBuildIR: false
worldCacheMode: false
scope: APP
nativeModel: true
planFile: null
analyses:
  pta: advanced:zipper-e;cs:ci;reflection-inference:null
onlyGenPlan: false
keepResult:
- $KEEP-ALL
  ```
</details>
<details>
  <summary>🔍 Click here to see Tai-e Analysis Plan</summary>
  <!-- Please leave one blank line below for enabling the code block rendering. -->

  ```yaml
  {{The content of 'output/tai-e-plan.yml' file}}
  ```
</details>


### 📜 Tai-e Log

<details>
  <summary>🔍 Click here to see Tai-e Log</summary>
  <!-- Please leave one blank line below for enabling the code block rendering. -->

  ```
  {{The content of 'output/tai-e.log' file}}
  ```
</details>


### ℹ️ Additional Information

_No response_

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -6,6 +6,9 @@
 - Pointer analysis
   - Add special handling for zero-length arrays to enhance PTA precision.
 
+### Fixes
+- Fix NPE in Zipper-e pre-analysis.
+
 ## [0.5.1] - 2024-12-31
 
 ### New Features

--- a/src/main/java/pascal/taie/analysis/pta/toolkit/zipper/Zipper.java
+++ b/src/main/java/pascal/taie/analysis/pta/toolkit/zipper/Zipper.java
@@ -186,7 +186,10 @@ public class Zipper {
         if (isExpress) {
             int accPts = 0;
             for (JMethod m : pcms) {
-                accPts += methodPts.get(m).intValue();
+                MutableInt mPtsSize = methodPts.get(m);
+                if (mPtsSize != null) {
+                    accPts += mPtsSize.intValue();
+                }
             }
             if (accPts > pcmThreshold) {
                 // clear precision-critical method group whose accumulative

